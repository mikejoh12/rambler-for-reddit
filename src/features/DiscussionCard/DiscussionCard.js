import Typography from '@material-ui/core/Typography';
import { makeStyles } from '@material-ui/core/styles';
import CardContent from '@material-ui/core/CardContent';
import Card from '@material-ui/core/Card';

const useStyles = makeStyles({
    root: {
      margin: 10
    },
    media: {
        maxWidth: 140
    },
  });

export const DiscussionCard = (props) => {
    const { post } = props;
    const classes = useStyles()

    return (
        <Card className={classes.root}>
                <CardContent>
                    <Typography variant="h6" fontStyle="italic" color="textSecondary">
                        {post.author}
                    </Typography>
                    <Typography variant="p" color="textSecondary">
                        {post.body}
                    </Typography>
                </CardContent>
        </Card>
    )
}