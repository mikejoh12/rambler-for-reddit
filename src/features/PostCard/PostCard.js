import React from 'react'
import Card from '@material-ui/core/Card';
import CardContent from '@material-ui/core/CardContent';
import CardMedia from '@material-ui/core/CardMedia';
import CardActionArea from '@material-ui/core/CardActionArea';
import Typography from '@material-ui/core/Typography';
import { makeStyles } from '@material-ui/core/styles';
import { Link } from 'react-router-dom'

const useStyles = makeStyles({
    root: {
      margin: 10
    },
    media: {
        maxWidth: 140
    },
  });

export const PostCard = (props) => {
    const { post } = props;
    const classes = useStyles()

    return (
        <Card className={classes.root}>
            <CardActionArea component={Link} to={`/discussion/${post.subreddit}/${post.id}`}>
                <CardMedia
                    component="img"
                    alt=""
                    image={post.imgUrl}
                    title={post.title}
                />

                {post.videoUrl && <CardMedia 
                    component="video"
                    height="200"
                    autoPlay
                    controls
                    image={post.videoUrl}
                    title={post.title}
                />}

                <CardContent>
                    <Typography variant="h6" color="textSecondary">
                        {post.title}
                    </Typography>
                    <Typography>
                        Posted by {post.author}
                    </Typography>
                    <Typography>
                        {`r/${post.subreddit}`}
                    </Typography>
                </CardContent>
            </CardActionArea>
        </Card>
    )
}