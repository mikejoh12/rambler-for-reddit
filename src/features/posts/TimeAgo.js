import React from 'react'
import { formatDistanceToNow } from 'date-fns'


export const TimeAgo = ({ timestamp }) => {
    let timeAgo = '';
    if (timestamp) {
        const unixTimestamp = timestamp
        const milliseconds = unixTimestamp * 1000
        const date = new Date(milliseconds)
        const timePeriod = formatDistanceToNow(date)
        timeAgo = `${timePeriod} ago`
    }

    return (
    <span title={timestamp}>
        &nbsp;{timeAgo}
    </span>
    )
}