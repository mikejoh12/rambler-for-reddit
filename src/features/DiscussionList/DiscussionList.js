import React, { useEffect } from 'react'
import { useParams } from 'react-router-dom'
import { useSelector, useDispatch } from 'react-redux'
import { DiscussionCard } from '../DiscussionCard/DiscussionCard'
import { selectDiscussion, fetchDiscussion } from '../reddit/redditSlice'
import { Typography } from '@material-ui/core'

export const DiscussionList = () => {

    let { id, subreddit } = useParams()
    
    const discussion = useSelector(selectDiscussion);
    const dispatch = useDispatch()

    //Fetch new posts
    useEffect(() => {
        dispatch(fetchDiscussion(`r/${subreddit}/${id}.json`))
    }, [subreddit, id, dispatch])

    let discussionList
    if (discussion) {
        discussionList = discussion.map(post => {
            return <DiscussionCard 
                        post={post}
                        key={post.id} />
        })
    }
        
    return (
            <div>
                <Typography variant="h6" color="textPrimary">
                    {`r/${subreddit}`}
                </Typography>
                {discussionList}
            </div>
    )
}