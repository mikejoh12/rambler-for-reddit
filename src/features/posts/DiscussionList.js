import React, { useEffect } from 'react'
import { useParams } from 'react-router-dom'
import { useSelector, useDispatch } from 'react-redux'
import { DiscussionCard } from './DiscussionCard'
import { selectDiscussion, fetchDiscussion, searchTargetUpdated, selectDiscussionStatus } from '../reddit/redditSlice'
import { Typography } from '@material-ui/core'
import CircularProgress from '@material-ui/core/CircularProgress';

export const DiscussionList = () => {

    let { id, subreddit } = useParams()
    
    const discussionStatus = useSelector(selectDiscussionStatus);
    const discussion = useSelector(selectDiscussion);
    const dispatch = useDispatch()

    useEffect(() => {
        dispatch(searchTargetUpdated('discussion'))
    }, [dispatch])

    //Fetch new posts
    useEffect(() => {
        dispatch(fetchDiscussion(`r/${subreddit}/${id}.json`))
    }, [subreddit, id, dispatch])
 
    return (
            <div>
                <Typography variant="h5" color="textPrimary">
                    {`r/${subreddit}`}
                </Typography>

                { (discussionStatus === 'loading') && <CircularProgress style={{marginTop: 30}} /> }

                { (discussion && discussionStatus === 'succeeded') && discussion.map(post => {
                    return <DiscussionCard 
                        post={post}
                        key={post.id} />
                    }) }
                
                { (!discussion.length && discussionStatus === 'succeeded') &&
                <Typography variant="h6" color="textPrimary">
                    No replies found.
                </Typography> }
            </div>
    )
}