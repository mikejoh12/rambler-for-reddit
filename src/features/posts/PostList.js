import React, { useEffect } from 'react'
import { PostCard } from './PostCard'
import { useSelector, useDispatch } from 'react-redux'
import { selectPosts, fetchPosts, selectPostsStatus, searchTargetUpdated } from '../reddit/redditSlice'
import { useParams } from 'react-router-dom'
import { Typography } from '@material-ui/core'
import CircularProgress from '@material-ui/core/CircularProgress';

export const PostList = () => {
    
    let { subreddit } = useParams()

    //Fetch new posts
    const dispatch = useDispatch()
    
    useEffect(() => {
        dispatch(searchTargetUpdated('posts'))
    }, [dispatch])

    useEffect(() => {
        dispatch(fetchPosts(subreddit))
    }, [subreddit, dispatch])
    
    const posts = useSelector(selectPosts)
    const postsStatus = useSelector(selectPostsStatus)

    return (
        <div>
            <Typography variant="h5" color="textPrimary">
                {`r/${subreddit}`}
            </Typography>

            {(postsStatus === 'loading') && <CircularProgress style={{marginTop: 30}} />}

            {(posts && postsStatus === 'succeeded') && posts.map(post => {
                return <PostCard 
                        post={post}
                        key={post.id} />
                })
            }       
        </div>
    )
}