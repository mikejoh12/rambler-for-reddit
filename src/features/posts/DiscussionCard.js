import Typography from '@material-ui/core/Typography';
import { makeStyles } from '@material-ui/core/styles';
import CardContent from '@material-ui/core/CardContent';
import Card from '@material-ui/core/Card';
import { TimeAgo } from './TimeAgo'
import SwapVertOutlinedIcon from '@material-ui/icons/SwapVertOutlined';
import { Grid } from '@material-ui/core'

const useStyles = makeStyles({
    root: {
      margin: 10,
      backgroundColor: '#D3E3F0'
    },
    media: {
        maxWidth: 140
    },
  });

export const DiscussionCard = (props) => {
    const { post } = props;
    const classes = useStyles()

    return (
        <Card className={classes.root}>
                <CardContent>
                    <Grid container direction="row" alignItems="center">
                        <SwapVertOutlinedIcon fontSize="small" />
                        <Typography variant="caption" align="left">
                                {post.ups}
                        </Typography>
                    </Grid>

                        <Typography variant="body1" color="textSecondary">
                            Posted by {post.author}
                            <TimeAgo timestamp={post.created_utc} />
                        </Typography>
                        

                    <Typography variant="body1" color="textPrimary">
                        {post.body}
                    </Typography>
                </CardContent>
        </Card>
    )
}