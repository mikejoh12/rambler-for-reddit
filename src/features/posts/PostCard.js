import React from 'react'
import Card from '@material-ui/core/Card';
import CardContent from '@material-ui/core/CardContent';
import CardMedia from '@material-ui/core/CardMedia';
import CardActionArea from '@material-ui/core/CardActionArea'
import Typography from '@material-ui/core/Typography'
import { makeStyles } from '@material-ui/core/styles'
import { Link } from 'react-router-dom'
import { TimeAgo } from './TimeAgo'
import SwapVertOutlinedIcon from '@material-ui/icons/SwapVertOutlined'
import ChatBubbleOutlineIcon from '@material-ui/icons/ChatBubbleOutline'
import { Grid } from '@material-ui/core'

const useStyles = makeStyles({
    root: {
      margin: 10,
      backgroundColor: '#D3E3F0',
    }
  });

export const PostCard = (props) => {
    const { post } = props;
    const classes = useStyles()

    const handleClick = () => {
        window.scrollTo(0, 0)
    }

    return (
    <Grid item>
    <Card className={classes.root}>                 
            {post.videoUrl && <CardMedia 
                component="video"
                height="200"
                autoPlay
                controls
                image={post.videoUrl}
                title={post.title}
            />}
             <CardActionArea component={Link} to={`/discussion/${post.subreddit}/${post.id}`} onClick={handleClick}>
                {post.post_hint === 'image' && <CardMedia
                    component="img"
                    alt=""
                    height="auto"
                    image={post.url}
                    title={post.title}
                    />
                }
            <CardContent>
                <Grid container direction="row" alignItems="center">
                    <SwapVertOutlinedIcon fontSize="small" />
                    <Typography variant="caption" align="left">
                            {post.ups}
                    </Typography>
                </Grid>
                
                <Typography variant="body1" color="textSecondary">
                    Posted by {post.author}
                    <TimeAgo timestamp={post.created_utc} />
                </Typography>
                <Typography variant="h6" color="textPrimary">
                    {post.title}
                </Typography>
                <Typography>
                    {`r/${post.subreddit}`}
                </Typography>

                <Grid container direction="row" alignItems="center">
                    <ChatBubbleOutlineIcon fontSize="small" />
                    <Typography variant="caption" align="left">
                            {post.num_comments}
                    </Typography>
                </Grid>
            </CardContent>
        </CardActionArea>
    </Card>
    </Grid> 
    )
}