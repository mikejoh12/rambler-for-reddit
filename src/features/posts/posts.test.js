
import { DiscussionCard } from './DiscussionCard'
import { DiscussionList } from './DiscussionList'
import { PostCard } from './PostCard'
import { PostList } from './PostList'
import { TimeAgo } from './TimeAgo'

import { Provider } from 'react-redux'
import configureStore from 'redux-mock-store'
const { shallow } = require("enzyme")

const mockStore = configureStore([])
const store = mockStore()

describe('DiscussionCard component', () => {
    it('should render with given state from Redux store', () => {
        const wrapper = shallow(
            <Provider store={store}>
                <DiscussionCard />
            </Provider>
        )
        expect(wrapper).toMatchSnapshot()
    })
})

describe('DiscussionList component', () => {
    it('should render with given state from Redux store', () => {
        const wrapper = shallow(
            <Provider store={store}>
                <DiscussionList />
            </Provider>
        )
        expect(wrapper).toMatchSnapshot()
    })
})

describe('PostCard component', () => {
    it('should render with given state from Redux store', () => {
        const wrapper = shallow(
            <Provider store={store}>
                <PostCard />
            </Provider>
        )
        expect(wrapper).toMatchSnapshot()
    })
})
  
describe('PostList component', () => {
    it('should render with given state from Redux store', () => {
        const wrapper = shallow(
            <Provider store={store}>
                <PostList />
            </Provider>
        )
        expect(wrapper).toMatchSnapshot()
    })
})
  
describe('TimeAgo component', () => {
    it('should render with given state from Redux store', () => {
        const wrapper = shallow(
            <Provider store={store}>
                <TimeAgo />
            </Provider>
        )
        expect(wrapper).toMatchSnapshot()
    })
})