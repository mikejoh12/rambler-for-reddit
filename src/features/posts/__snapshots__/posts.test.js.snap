// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DiscussionCard component should render with given state from Redux store 1`] = `ShallowWrapper {}`;

exports[`DiscussionList component should render with given state from Redux store 1`] = `ShallowWrapper {}`;

exports[`PostCard component should render with given state from Redux store 1`] = `ShallowWrapper {}`;

exports[`PostList component should render with given state from Redux store 1`] = `ShallowWrapper {}`;

exports[`TimeAgo component should render with given state from Redux store 1`] = `ShallowWrapper {}`;
